File: 프로그래머스/0/181927. 마지막 두 원소/마지막 두 원소.py
def solution(num_list):
    answer = []
    
    # num_list 길이만큼 for문 돌리자
    for i in range(len(num_list)):
        answer.append(num_list[i])  # [2, 1, 6]
        # print(i)
        
    if(answer[-1] > answer[-2]):    # 6이 1보다 크다 (마지막 원소가 클 때)
        answer.append(answer[-1] - answer[-2])        # 뺀 값
    else:   # 마지막 원소가 작을 때
        answer.append(answer[-1] * 2)   # 마지막 원소 두 배 값
    return answer

File: 프로그래머스/0/181927. 마지막 두 원소/마지막 두 원소.java
class Solution {
    public int[] solution(int[] num_list) {
        int[] answer = new int[num_list.length+1];
        int index = 0;
        
        for(int i=0; i<num_list.length; i++) {
            answer[index] = num_list[i];
            index++;
        }
        
        if(num_list[num_list.length-1] > num_list[num_list.length-2]) { // 마지막 원소가 클 때
            answer[index] = num_list[num_list.length-1] - num_list[num_list.length-2];
        } else {    // 마지막 원소가 작을 때
            answer[index] = num_list[num_list.length-1] * 2;
        }
        return answer;
    }
}